### PR TITLE
[msal-extensions] Extensions #5

### PR DESCRIPTION
**Pull request by sangonzal**
- Add logger
- Update README and .gitignore

### CODE REVIEW

File: extensions/src/persistence/FilePersistence.ts [hunk 1]
```diff
@@ -9,14 +9,20 @@ import { dirname } from "path";
...
+import { Logger, LoggerOptions, LogLevel } from "@azure/msal-common";
 
 export class FilePersistence implements IPersistence {
 
     private filePath: string;
+    private logger: Logger;
 
-    public static async create(fileLocation: string): Promise<FilePersistence> {
+    private constructor() {
+    }
```
Comment (jasonnutter) [inline extensions/src/persistence/FilePersistence.ts:20]: Not needed if not used.

File: extensions/README.md [hunk 2]
```diff
@@ -16,17 +16,15 @@ Linux - LibSecret is used for encryption.
 
 ## Building
 
-The extensions contain prebuild binaries. To build from source, you will need Python on you path.
+The extensions contain prebuild binaries. To build from source, you will need Python on you path,
+as [node-gyp](https://github.com/nodejs/node-gyp) is used to build addons for accessing system APIs.
+
+On linux, the library uses `libsecret` so you may need to install it.
```
Comment (jasonnutter) [inline extensions/README.md:22]: I would include or link to instructions for how to check that it is installed and installed it if not.

File: extensions/src/persistence/FilePersistence.ts (hunk 1 above)
Comment (sangonzal) [reply extensions/src/persistence/FilePersistence.ts]: Removed

File: extensions/README.md (hunk 2 above)
Comment (sangonzal) [reply extensions/README.md]: Added instructions for a couple of distributions.

**Comment by sameerag** [comment]
One general comment: Dependency on `msal-common` seems to be hardcoded. We need to have a strategy regarding updating `msal-common` across libraries.

**Pull request merged**